h/static/scripts/directive/annotation.js

```diff
@@ -0,0 +1,730 @@
+      references = [references];
+    }
+
+    references = references.concat(id);
+
+    reply = annotationMapper.createAnnotation({
+      references: references,
+      uri: uri
+    });
+    reply.group = model.group;
+
+    if (session.state.userid) {
+      if (permissions.isShared(model.permissions, model.group)) {
+        reply.permissions = permissions.shared(reply.group);
+      } else {
+        reply.permissions = permissions.private();
+      }
+    }
+  };
+
+  /**
+    * @ngdoc method
+    * @name annotation.AnnotationController#render
+    * @description Called to update the view when the model changes.
+    */
+  vm.render = function() {
+    var documentTitle;
+    var domain;
+    var draft = drafts.get(model);
+    var i;
+    var link;
+    var tagsAsObjects;
+    var uri = model.uri;
+
+    // Extend the view model with a copy of the domain model.
+    // Note that copy is used so that deep properties aren't shared.
+    vm.annotation = angular.extend({}, angular.copy(model));
+
+    // If we have unsaved changes to this annotation, apply them
+    // to the view model.
+    if (draft) {
+      angular.extend(vm.annotation, angular.copy(draft));
+    }
+
+    vm.annotationURI = new URL(
+      '/a/' + vm.annotation.id, vm.baseURI).href;
+
+    // Extract the document metadata.
+    domain = new URL(uri).hostname;
+    if (model.document) {
+      if (uri.indexOf('urn') === 0) {
+        for (i = 0; i < model.document.link.length; i++) {
+          link = model.document.link[i];
+          if (!(link.href.indexOf('urn'))) {
+            continue;
+          }
+          uri = link.href;
+          break;
+        }
+      }
+
+      documentTitle = Array.isArray(
+        model.document.title) ? model.document.title[0] : model.document.title;
+
+      vm.document = {
+        uri: uri,
+        domain: domain,
+        title: documentTitle || domain
+      };
+    } else {
+      vm.document = {
+        uri: uri,
+        domain: domain,
+        title: domain
+      };
+    }
+
+    if (vm.document.title.length > 30) {
+      vm.document.title = vm.document.title.slice(0, 30) + '…';
+    }
+
+    // Form the tags for ngTagsInput.
+    tagsAsObjects = [];
+    for (i = 0; i < (vm.annotation.tags || []).length; i++) {
+      tagsAsObjects[tagsAsObjects.length] = {text: vm.annotation.tags[i]};
+    }
+    vm.annotation.tags = tagsAsObjects;
+  };
+
+  updateTimestamp = function(repeat) {
+    var fuzzyUpdate;
+    var nextUpdate;
+
+    repeat = repeat || false;
+
+    // New (not yet saved to the server) annotations don't have any .updated
+    // yet, so we can't update their timestamp.
+    if (!model.updated) {
+      return;
+    }
+
+    vm.timestamp = time.toFuzzyString(model.updated);
+
+    if (!repeat) {
+      return;
+    }
+
+    fuzzyUpdate = time.nextFuzzyUpdate(model.updated);
+    nextUpdate = (1000 * fuzzyUpdate) + 500;
+
+    $timeout(function() {
+      updateTimestamp(true);
+      $scope.$digest();
```

Comment:
OK, so presumably the idea here is to set `$timeout`'s 'apply' parameter to false to avoid a `$rootScope.$apply` and instead only digest the current scope. That's a sensible thing to do from a performance point of view, but could use a comment.

In general, I think it is a good idea to put a comment after any non-obvious boolean parameters to note what they are for, since JS doesn't have keyword arguments (eg. `$timeout(..., false /* apply */)`)